stdlib/public/core/Array.swift

```diff
@@ -1842,7 +1842,15 @@ extension Array {
   ///     numbers.withUnsafeBytes {
   ///         byteBuffer.append(contentsOf: $0)
   ///     }
-  ///     // byteBuffer == [1, 0, 0, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, ...]
+  ///     // On 64-bit little-endian machine:
```

Comment:
Let's do the same thing here as for the example above.

First, let's take word size out of the equation, because it is irrelevant to this documentation. Instead, let's make `numbers` of type `[Int32]` just like another person did above. Then, let's take the information about endianness out of the comment-within-a-comment and make it a real link, and put that in the introductory remarks, something like:

> The following example copies the bytes of the `numbers` array into a buffer of `UInt8`; on a [little-endian](#) machine, the result is as shown:
> [...]
> `// byteBuffer == [1, 0, 0, 0, 2, 0, 0, 0, 3, 0, 0, 0]`